cmd/prometheus/main.go

```diff
@@ -152,6 +155,7 @@ type flagConfig struct {
 	enableNewSDManager         bool
 	enablePerStepStats         bool
 	enableAutoGOMAXPROCS       bool
+	enableKubeSecretProvider   bool
```

Comment:
Nit, but probably feature flag would be for secret provider generally. E.g. we don't have feature flag for every SD provider added.

```suggestion
	enableSecretProvider   bool
```

Reply:
That's true, and this is intended. The idea is that when the Prometheus upstream version is eventually merged, it will not be compatible with this one. We may want to keep both implementations until we update to the upstream version within prometheus-engine.

Reply:
IIUC you have a different name for to avoid a merge conflict in the future? If so, that seems like it may actually lead to more confusion during resolving conflicts, not less, right?

Reply:
Can you elaborate on how it could lead to more confusion?

Reply:
Meh - it's really a speculative (non-blocking) nit.

I was thinking if you have a flag `--enableGMPSecretProvider` and upstream they name it `--enableSecretProvider`, but the two implementations end up doing similar things, the merging that happens may actually end up doubling code in a weird way (the merge thinks of them as two separate code paths).

Whereas if you happened to have the same name as the upstream version, it would legitimately try to interlace the code paths in a way that makes more sense, and having to manually resolve that interlace could end up being more straightforward.

I guess my point is: trying to proactively avoid future merge conflicts is too unpredictable, and you should just do what's easiest here.